DOC-Modified documentation for the issue GH42106 (#42110)
pandas-dev/pandas: doc/source/user_guide/indexing.rst

----------------------------------------

Sometimes you want to extract a set of values given a sequence of row labels

> and column labels, this can be achieved by ``pandas.factorize`` which extracts the distinct values of the intended column and can be indexed by passing the length of the dataframe to the 
> ``numpy.arange`` function and the distinct value array.  For instance:

Reviewer: this is very complicated wording...how about
```suggestion
and column labels, this can be achieved by using ``pandas.factorize`` and NumPy indexing. For instance:
```

Author: I have modified the file as per your suggestion.

                       'A': [80, 23, np.nan, 22],
                       'B': [80, 55, 76, 67]})
    df
    idx, cols = pd.factorize(df['col'])

>     df.reindex(cols, axis=1).to_numpy()[np.arange(len(df)), idx]

Reviewer 2: The text above the example will also need to be changed?

Author: Yes, you are right. I have modified the text which is there above the example,as per my understanding. Can you review it.

Reviewer 3: show the result in the PR itself as this doesn't look to replicate

Reviewer: On pandas 1.1.5:
```python
In [1]:    df = pd.DataFrame({'col': ["A", "A", "B", "B"],
   ...:    ...:                        'A': [80, 23, np.nan, 22],
   ...:    ...:                        'B': [80, 55, 76, 67]})
   ...: 

In [2]: df.lookup(df.index, df['col'])
Out[2]: array([80., 23., 76., 67.])
```

the example on master:
```python
In [2]:     melt = df.melt('col')
   ...:     melt = melt.loc[melt['col'] == melt['variable'], 'value']
   ...:     melt.reset_index(drop=True)
Out[2]: 
0    80.0
1    23.0
2    76.0
3    67.0
Name: value, dtype: float64
```

the example here:
```python
In [3]:     idx, cols = pd.factorize(df['col'])
   ...:     df.reindex(cols, axis=1).to_numpy()[np.arange(len(df)), idx]
Out[3]: array([80., 23., 76., 67.])
```
